web/src/utils/externalServices.ts

```diff
@@ -0,0 +1,184 @@
+  if (typeof endpoint == 'string') {
+    resolvedEndpoint = endpoint;
+  } else if (typeof endpoint == 'function') {
+    resolvedEndpoint = endpoint(data);
+  } else {
+    throw new Error('Invalid endpoint type');
+  }
+
+  if (!resolvedEndpoint) {
+    return null;
+  }
+
+  return resolvedEndpoint
+    .replaceAll('$dandiset_id$', data.dandisetId)
+    .replaceAll('$dandiset_version$', data.dandisetVersion)
+    .replaceAll('$asset_url$', data.assetUrl)
+    .replaceAll('$asset_dandi_url$', data.assetDandiUrl)
+    .replaceAll('$asset_s3_url$', data.assetS3Url);
+}
+
+export function getExternalServices(path: AssetPath, info: {dandisetId: string, dandisetVersion: string}) {
+  if (path.asset === null) {
+    return [];
+  }
+
+  const store = useDandisetStore();
+  const embargoed = computed(() => store.dandiset?.dandiset.embargo_status === 'EMBARGOED');
+
+  const servicePredicate = (service: ExternalService, _path: AssetPath) => (
+    new RegExp(service.regex).test(path.path)
+          && _path.asset !== null
+          && _path.aggregate_size <= service.maxsize
+  );
+
+  // Formulate the two possible asset URLs -- the direct S3 link to the relevant
+  // object, and the DANDI URL that redirects to the S3 one.
+  const baseApiUrl = import.meta.env.VITE_APP_DANDI_API_ROOT;
+  const assetDandiUrl = `${baseApiUrl}assets/${path.asset?.asset_id}/download/`;
+  const assetS3Url = trimEnd((path.asset as AssetFile).url, '/');
+  const assetId = path.asset?.asset_id;
+
+  if (!assetId) {
+    // This should never happen
+    throw new Error('Asset ID is not defined');
+  }
+
+  // Select the best "default" URL: the direct S3 link is better when it can be
+  // used, but we're forced to supply the internal DANDI URL for embargoed
+  // dandisets (since the ready-made S3 URL will prevent access in that case).
+  const assetUrl = embargoed.value ? assetDandiUrl : assetS3Url;
+
+  return EXTERNAL_SERVICES
+    .filter((service) => servicePredicate(service, path))
+    .flatMap((service) => {
+      const url = serviceURL(service.endpoint, {
+        dandisetId: info.dandisetId,
+        dandisetVersion: info.dandisetVersion,
+        assetId,
+        assetUrl,
+        assetDandiUrl,
+        assetS3Url,
+      });
+      return url ? [{ name: service.name, url }] : [];
+    });
+}
+
+/**
+ * Custom function used to generate the endpoint
+ * for the "Neuroglancer" service
+ */
+function redirectNeuroglancerUrl(item: ServiceUrlData): string | null {
+  const store = useDandisetStore();
+  const embargoed = computed(() => store.dandiset?.dandiset.embargo_status === 'EMBARGOED');
+  if (embargoed.value) {
+    return null;
+  }
+
+  const assetS3Url = trimEnd(item.assetS3Url, '/');
+  const baseUrl = `https://neuroglancer-demo.appspot.com/#!`; // Public neuroglancer instance
+  const jsonObject = {
+    layers: [
+      {
+        type: "new",
+        source: assetS3Url.includes("zarr") ? `zarr://${assetS3Url}` : `nifti://${assetS3Url}`,
```

Comment:
Perhaps this would work since the `https://` prefix is now used in Neuroglancer for zarr and nifti.

```suggestion
        source: assetS3Url.includes("zarr") ? `${assetS3Url}` : `${assetS3Url}`,
```

Reply:
Seems this didn't work. see https://deploy-preview-2063--sandbox-dandiarchive-org.netlify.app/dandiset/202598/draft/files?location=sub-01/func&page=1

I'll grab some of these files off of production and try to get this working locally